### PR TITLE
SUBMARINE-805. Install local traefik chart via helm golang API

### PR DESCRIPTION
**Pull request by kevin85421**
### What is this PR for?
Install local traefik chart via Helm golang API.

traefik chart: https://github.com/apache/submarine/tree/master/helm-charts/submarine/charts/traefik

Helm golang API: https://github.com/apache/submarine/blob/master/submarine-cloud-v2/pkg/helm/helm.go

### What type of PR is it?
[Feature]

### Todos
* port-forwarding via client-go

### What is the Jira issue?
https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-805?filter=allissues#

### How should this be tested?
```
# initialization
# Step1: Register Custom Resource Definition
kubectl apply -f artifacts/examples/crd.yaml

# Step2: Create a Custom Resource
kubectl apply -f artifacts/examples/example-submarine.yaml


# Step3-1: helm
cd submarine
helm install submarine ./helm-charts/submarine
kubectl port-forward --address 0.0.0.0 service/traefik 32080:80

# Step3-2: in-cluster
cd submarine/submarine-cloud-v2
eval $(minikube docker-env)
make image
kubectl apply -f artifacts/examples/submarine-operator-service-account.yaml
kubectl apply -f artifacts/examples/submarine-operator.yaml
kubectl port-forward --address 0.0.0.0 service/traefik 32080:80

# Step3-3: out-of-cluster
go build -o submarine-operator
./submarine-operator
kubectl port-forward --address 0.0.0.0 service/traefik 32080:80

# Step4: Check 127.0.0.1:32080 in web browser
#.            Username: admin / Password: admin
```

### Screenshots (if appropriate)

https://user-images.githubusercontent.com/20109646/116415503-10663480-a86c-11eb-8fff-199cc69d31c7.mov

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

**Comment by kevin85421** [comment]
@MortalHappiness can you help me review this PR?

File: submarine-cloud-v2/Makefile [hunk 1]
```diff
@@ -28,6 +28,9 @@ api:
 
 .PHONY: image
 image:
+ifeq (,$(wildcard ./charts/))
+	cp -r ../helm-charts/submarine/charts  .
+endif
 	GOOS=linux go build -o submarine-operator
 	docker build -t submarine-operator .
 	go build -o submarine-operator
```
Comment (MortalHappiness) [inline submarine-cloud-v2/Makefile:36]: You only copy `../helm-charts/submarine/charts` to current directory when we run `make image`. Therefore, if we run `make` without running `make image`, the `charts` directory will not exist and out-of-cluster operator will throw an error.
Maybe a better workaround is making the `charts` folder a non-phony target and let `make` and `make image` dependent on it
```make
all: # The default target

charts:
	cp -r ../helm-charts/submarine/charts .

.PHONY: all
all: charts
	go build -o submarine-operator

.PHONY: image
image: charts
	GOOS=linux go build -o submarine-operator
	docker build -t submarine-operator .
	go build -o submarine-operator
```

**Review by MortalHappiness — CHANGES_REQUESTED** [review]

File: submarine-cloud-v2/Makefile (hunk 1 above)
Comment (kevin85421) [reply submarine-cloud-v2/Makefile]: Thanks! I have updated it.

**Comment by MortalHappiness** [comment]
LGTM

**Pull request merged**